src/imgix-core-js.js

```diff
@@ -174,10 +174,16 @@
         targetWidths = this._generateTargetWidths(widthTolerance, minWidth, maxWidth);
       }
 
+      var queryParams = {};
+      var key;
+      for (key in params) {
+        queryParams[key] = params[key];
+      }
+
       for (var i = 0; i < targetWidths.length; i++) {
         currentWidth = targetWidths[i];
-        params.w = currentWidth;
-        srcset += this.buildURL(path, params) + ' ' + currentWidth + 'w,\n';
+        queryParams.w = currentWidth;
+        srcset += this.buildURL(path, queryParams) + ' ' + currentWidth + 'w,\n';
       }
 
       return srcset.slice(0,-2);
```

Comment:
Nit: I know this outside of the scope of this change but what if we initialized the local var `srcset` as an array and then `.join("\n")` at the end?

Reply:
Yeah, I plan to do that, but didn't because it was outside the scope of this change/issue. 

This is something I've had my eye on for a while, but planned to follow this PR with another to make these changes.

However, I am more than happy to make them within this PR though. Just lmk :)